### PR TITLE
Forms Automated Test - LPS-133520

### PR DESCRIPTION
**Pull request by wesleyroberts**

### CODE REVIEW

**Comment by wesleyroberts** [comment]
ci:test:forms

**Comment by wesleyroberts** [comment]
ci:test:forms

**Comment by wesleyroberts** [comment]
ci:test:relevant

**Comment by wesleyroberts** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase [hunk 1]
```diff
@@ -255,14 +255,46 @@ definition {
...
 
+		FormPortletConfiguration.viewTranslationLanguage(symbolName = "en-US");
+
+		FormFields.viewLabel(
+			fieldLabel = "Text",
+			fieldName = "Text");
+
+		FormPortletConfiguration.changeLanguageTo(changeLanguageTo = "en-US");
+
+		FormPortletConfiguration.changeLanguageTo(changeLanguageTo = "español-España");
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase:293]: @wesleyroberts, apparently, the macro `Form.selectTranslation` can't help you here, right?

In addition, I realized that the following pattern isn't being used, is there any special reason for it? If not, can you please use the pattern?

```
var key_changeLanguageTo = "${changeLanguageTo}";
```

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro [hunk 2]
```diff
@@ -31,4 +35,10 @@ definition {
 			value1 = "Displaying Form: ${formName} (Modified)");
 	}
 
+	macro viewTranslationLanguage {
+		AssertElementPresent(
+			changeLanguageTo = "${symbolName}",
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro:40]: @wesleyroberts, why did you decide to name the variable `symbolName`?

How about using a name more related to language, like  `translationLanguage` or `language` only?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase [hunk 3]
```diff
@@ -255,14 +255,46 @@ definition {
...
+
+		Navigator.selectSpecificWindow(windowTitle = "Form - Forms");
 
+		FormPortletConfiguration.viewTranslationLanguage(symbolName = "en-US");
+
+		FormFields.viewLabel(
+			fieldLabel = "Text",
+			fieldName = "Text");
+
+		FormPortletConfiguration.changeLanguageTo(changeLanguageTo = "en-US");
```
[the remark this answers is not in the document]
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase:291]: you're right, this step it's not necessary, thanks for warning me!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase (hunk 1 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase:293]: I can use it, I didn't make the macro, just take advantage of a co-worker

Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase]: Changes made

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro (hunk 2 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro]: Changes made

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase (hunk 1 above)
Comment (natocesarrego) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase]: @wesleyroberts, the pattern is still different. :disappointed: Please see my previous comment.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormPortletConfiguration.path [hunk 4]
```diff
@@ -14,6 +14,12 @@
 	<td>xpath=(//div[contains(@id,'SearchContainer')]//tr[contains(.,'${key_formName}')]/td//a)[1]</td>
 	<td></td>
 </tr>
+<!--TRANSLATE-->
+<tr>
+	<td>SELECT_LANGUAGE</td>
+	<td>//span[contains(text(),'${changeLanguageTo}')]</td>
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormPortletConfiguration.path:20]: @wesleyroberts, please rename:

- `changeLanguageTo` to `key_language`

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro [hunk 5]
```diff
@@ -31,4 +40,10 @@ definition {
 			value1 = "Displaying Form: ${formName} (Modified)");
 	}
 
+	macro viewTranslationLanguage {
+		AssertElementPresent(
+			changeLanguageTo = "${language}",
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro:45]: @wesleyroberts, please rename:

- `changeLanguageTo` to `key_language`

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro [hunk 6]
```diff
@@ -2,6 +2,15 @@ definition {
 
 	var formName = "Created Form Name";
 
+	macro changeLanguageTo {
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro:5]: @wesleyroberts, please rename:

- `changeLanguageTo` to `changeLanguage`

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro [hunk 7]
```diff
@@ -2,6 +2,15 @@ definition {
...
+		Click(
+			changeLanguageTo = "${getLanguage}",
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro:7]: @wesleyroberts, please rename:

- `changeLanguageTo` to `key_language`
- `getLanguage` to `currentLanguage`

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro [hunk 8]
```diff
@@ -2,6 +2,15 @@ definition {
...
+			locator1 = "FormPortletConfiguration#SELECT_LANGUAGE");
+		Click(
+			changeLanguageTo = "${changeLanguageTo}",
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro:10]: @wesleyroberts, please rename:

- `changeLanguageTo` to `key_language`
- `${changeLanguageTo}` to `${newLanguage}`

**Review by natocesarrego — CHANGES_REQUESTED** [review]

**Comment by natocesarrego** [comment]
@wesleyroberts, sorry for ask you more renaming, but I've felt that indeed it's necessary and important. Please go ahead and refactor the other tests using the macro `changeLanguageTo`. Please follow the pattern `var key_changeLanguageTo = "${changeLanguageTo}";`.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase (hunk 1 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsTranslations.testcase]: Sorry for the mistakes, the changes were made

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormPortletConfiguration.path (hunk 4 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormPortletConfiguration.path]: changes made

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro (hunk 5 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro]: changes made

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro (hunk 7 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro]: changes made

File: portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro (hunk 8 above)
Comment (wesleyroberts) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormPortletConfiguration.macro]: changes made

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**